eva/readers/decord_reader.py

```diff
@@ -51,16 +56,29 @@ def __init__(
             sampling_rate (int, optional): Set if the caller wants one frame
             every `sampling_rate` number of frames. For example, if `sampling_rate = 10`, it returns every 10th frame. If both `predicate` and `sampling_rate` are specified, `sampling_rate` is given precedence.
             sampling_type (str, optional): Set as IFRAMES if caller want to sample on top on iframes only. e.g if the IFRAME frame numbers are [10,20,30,40,50] then'SAMPLE IFRAMES 2' will return [10,30,50]
+            read_audio (bool, optional): Whether to read audio stream from the video. Defaults to False
+            read_video (bool, optional): Whether to read video stream from the video. Defaults to True
         """
         self._predicate = predicate
         self._sampling_rate = sampling_rate or 1
         self._sampling_type = sampling_type
+        self._read_audio = read_audio
+        self._read_video = read_video
         super().__init__(*args, **kwargs)
 
     def _read(self) -> Iterator[Dict]:
         decord = _lazy_import_decord()
-        video = decord.VideoReader(self.file_url)
-        num_frames = int(len(video))
+        if self._read_audio:
```

Comment:
nit: we can move this logic (reader selection) to a separate function. Just want to clean the code as much as possible.